chunky/src/java/se/llbit/chunky/ui/render/LightingTab.java

```diff
@@ -108,16 +113,30 @@ public LightingTab() throws IOException {
 
     emitterSamplingStrategy.getItems().addAll(EmitterSamplingStrategy.values());
     emitterSamplingStrategy.getSelectionModel().selectedItemProperty()
-            .addListener((observable, oldvalue, newvalue) -> {
-              scene.setEmitterSamplingStrategy(newvalue);
-              if(newvalue != EmitterSamplingStrategy.NONE && scene.getEmitterGrid() == null)
-                // TODO add nice UI dialog for this
-                Log.warn("The world needs to be reloaded for emitter sampling to work");
-            });
+        .addListener((observable, oldvalue, newvalue) -> {
+          scene.setEmitterSamplingStrategy(newvalue);
+          if (newvalue != EmitterSamplingStrategy.NONE && scene.getEmitterGrid() == null) {
+            Alert warning = Dialogs.createAlert(AlertType.CONFIRMATION);
+            warning.setContentText("The selected chunks need to be reloaded in order for emitter sampling to work.");
+            warning.getButtonTypes().setAll(
+              ButtonType.CANCEL,
+              new ButtonType("Reload chunks", ButtonData.FINISH));
+            warning.setTitle("Chunk reload required");
+            ButtonType result = warning.showAndWait().orElse(ButtonType.CANCEL);
+            if (result.getButtonData() == ButtonData.FINISH) {
+              controller.getRenderController().getSceneManager().reloadChunks();
+            } else {
+              scene.setEmitterSamplingStrategy(EmitterSamplingStrategy.NONE);
+              emitterSamplingStrategy.getSelectionModel().select(EmitterSamplingStrategy.NONE);
```

Comment:
Maybe make it clear in the text of the button that emitter sampling will be disabled or don't disable emitter sampling. Maybe the user wants to do several changes that would need the chunks to be reloaded (like y min/max) but doesn't want to reload for every change

Reply:
Good point! Didn't think of that